File: getPowerNumber.js

function checkIfPowerNumber3(num){
    limit = parseInt(Math.sqrt(num))+1
    for (var x=2; x<limit; x++){
        b = Math.log(num)/Math.log(x);
        // console.log(b)
        b = precisionRound(b, 10) 
        if (b == parseInt(b)){
            return true
        }
    }
    return false
  }
  
  function precisionRound(number, precision) {
   var factor = Math.pow(10, precision);
   console.log("precision" , Math.round(number * factor) / factor);
   return Math.round(number * factor) / factor;
   
  }
  



  function getPowerNumber(index){
    var list_nums = []
    var start = 4
    while ((list_nums.length) != index+1) {
        if (checkIfPowerNumber3(start)){
            list_nums.push(start)
        }
        start+=1
    }  
    // console.log(list_nums)
    return list_nums[index]
    
  }
  console.log(getPowerNumber(2))

File: getPowerNumber.py
import math
# def checkIfPowerNumber(num):
#     limit = int(math.sqrt(num))+1
#     for x in xrange(2,limit):
#         p = x
#         while p <= num:
#             p = p * x
#             if p == num:
#                 return True
#     return False       
 
# def checkIfPowerNumber2(num):
#     limit = int(math.sqrt(num))+1
#     for x in xrange(2,limit):
#         b = math.floor(num**(1/float(x)))
#         # print 1/float(num),b
#         if b**x == num:
#             return True
#     return False


def check(num):
    limit = int(math.sqrt(num))+1
    for x in xrange(2,limit):
        b = math.log(num,x)
        if b == int(b):
        	return True
    return False    
 
def getPowerNumber(index):
    listNums = []
    start = 4
    while len(listNums) != index+1:
        if check(start):
            # print start
            listNums.append(start)
        start+=1
    # print start   
    return listNums[index]
 
print getPowerNumber(10)
